### DISCUSSION TITLE
Maximize window with gaps.

### DISCUSSION THREAD

**Discussion by lmeerwood** [RamonUnch/AltDrag#425] [Ideas]
I love using the gap feature with AltSnap. Generally I'll find myself dragging the window size to maximum so that it snaps around the edges.

Here's a screenshot showing what I mean:

![image](https://user-images.githubusercontent.com/6085575/232347244-f451ceea-e25d-4c8e-90b3-cf87f8533d86.png)

When you alt+double click to maximize it doesn't keep the gaps. Like so:

![image](https://user-images.githubusercontent.com/6085575/232347282-a7b7d8fa-6a2c-4b00-b19c-6d5c25b351da.png)

From a (slightly) educated guess it would be because double clicking just calls the normal system maximize function call which is completely unaware of AltSnap or its settings. 

Would there be a way to get AltSnap to maximize a window while respecting the gap settings? I know this wouldn't be a true window maximize.

**Comment by RamonUnch** [comment]
Indeed as you guessed the Maximize option use native maximization method.

For now it is possible to simulate a pseudo-maximize by maximizing vertically and horizontally (you can do it with Alt+Double RightClick Alt+Shift+Double RightClick in the central region of the window)

It would be a nice action to add indeed.

**Reply by lmeerwood** [reply]
Hey that's perfect. I'm happy to use that shortcut.
